libusb/descriptor.c

```diff
@@ -1242,7 +1253,7 @@ static int parse_iad_array(struct libusb_context *ctx,
 		// Second pass: Iterate through desc list, fill IAD structures
 		consumed = 0;
 		i = 0;
-		while (consumed < size) {
+		while (consumed + 9 < size) {
```

Comment:
Prevents nine byte eight heap overflow in the case that the amount consumed less than size, but by less than the descriptor we're about to read.